### PR TITLE
[IMP] Implements name_get() in 'project.task'

### PR DESCRIPTION
**Pull request by rodrigets**
Relational fields and the task notification email inform the task code.

email

m2o

### CODE REVIEW

**Review by aldo774 — APPROVED** [review]
I think that is a valid implementation :+1:

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
Don't overwrite the method without calling super, as you are breaking inheritance. Call super, and on the result, append at the beginning the code.

**Review by mstuttgart — APPROVED** [review]

File: project_task_code/models/project_task.py [hunk 1]
```diff
@@ -27,3 +27,15 @@ def copy(self, default=None):
...
+
+    @api.multi
+    @api.depends('name', 'code')
+    def name_get(self):
+        result = super(ProjectTask, self).name_get()
+        new_result = []
+
+        for task in result:
+            rec = self.browse(task[0])
+            name = '%s - %s' % (rec.code, rec.name)
```
Comment (tarteo) [inline project_task_code/models/project_task.py]: Can you change this to:

`name = '%s - %s' % (rec.code, task[1])`

So other modules can still append other strings to it.

**Review by tarteo — CHANGES_REQUESTED** [review]

**Comment by rodrigets** [comment]
Thank you @tarteo , this review is really relevant.
I made the change.

**Review by tarteo — APPROVED** [review]
@rodrigets Thanks!

File: project_task_code/models/project_task.py [hunk 2]
```diff
@@ -27,3 +27,15 @@ def copy(self, default=None):
             default = {}
         default['code'] = self.env['ir.sequence'].next_by_code('project.task')
         return super(ProjectTask, self).copy(default)
+
+    @api.multi
+    @api.depends('name', 'code')
```
Comment (dreispt) [inline project_task_code/models/project_task.py]: The api.depends is not relevant for name_get, you should remove it.

File: project_task_code/models/project_task.py [hunk 3]
```diff
@@ -27,3 +27,15 @@ def copy(self, default=None):
...
+    def name_get(self):
+        result = super(ProjectTask, self).name_get()
+        new_result = []
+
+        for task in result:
+            rec = self.browse(task[0])
+            name = '%s - %s' % (rec.code, task[1])
```
Comment (dreispt) [inline project_task_code/models/project_task.py]: Do you have a strong opinion about this format?
In other modules I used the `[%s] %s` format instead.

Comment (pedrobaeza) [reply project_task_code/models/project_task.py]: +1 to this

**Comment by rodrigets** [comment]
@dreispt , I have not strong a opinion for format, i go utilize your suggestion.
Thank you.

**Review by dreispt — CHANGES_REQUESTED** [review]
Waiting for fixes.

**Comment by rodrigets** [comment]
@dreispt , which fixes?

**Comment by dreispt** [comment]
Sorry - they were already there.

**Review by dreispt — APPROVED** [review]

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**